File: Core CS/Algorithms/Implementation/Drawing Book/drawingBook.scala
import java.io.PrintWriter

object Solution {

    def pageCount(n: Int, p: Int): Int = {
        return Array(p/2,n/2-p/2).min
    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val n = stdin.readLine.trim.toInt

        val p = stdin.readLine.trim.toInt

        val result = pageCount(n, p)

        printWriter.println(result)

        printWriter.close()
    }
}

File: Core CS/Algorithms/Implementation/Drawing Book/drawingBook.kt
import java.io.*
import java.text.*
import java.util.*
import kotlin.math.*

fun pageCount(n: Int, p: Int): Int {
    return min(p/2,n/2-p/2)
}

fun main(args: Array<String>) {
    val scan = Scanner(System.`in`)

    val n = scan.nextLine().trim().toInt()

    val p = scan.nextLine().trim().toInt()

    val result = pageCount(n, p)

    println(result)
}
